File: DYAlertPickerViewDemo/ViewController.m
//
//  ViewController.m
//  DYAlertPickerViewDemo
//
//  Created by danny on 2015/7/7.
//  Copyright (c) 2015年 danny. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property NSArray *item;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.item = @[@"Item 1", @"Item 2", @"Item 3", @"Item 4", @"Item 5"];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (NSAttributedString *)DYAlertPickView:(DYAlertPickView *)pickerView
                           titleForRow:(NSInteger)row{
    NSAttributedString *str = [[NSAttributedString alloc] initWithString:self.item[row]];
    return str;
}
- (NSInteger)numberOfRowsInDYAlertPickerView:(DYAlertPickView *)pickerView {
    return self.item.count;
}
- (void)DYAlertPickView:(DYAlertPickView *)pickerView didConfirmWithItemAtRow:(NSInteger)row{
    NSLog(@"%@ didConfirm", self.item[row]);
}

- (void)DYAlertPickerViewDidClickCancelButton:(DYAlertPickView *)pickerView {
    NSLog(@"Canceled");
}

- (void)DYAlertPickerViewDidClickSwitchButton:(DYAlertPickView *)pickerView switchButton:(UISwitch *)switchButton {
    NSLog(@"switch:%@",(switchButton.isOn?@"On":@"Off"));
}


- (BOOL)DYAlertPickerViewStateOfSwitchButton {
    return YES;
}

- (IBAction)showAlertPickerView:(id)sender {
    DYAlertPickView *picker = [[DYAlertPickView alloc] initWithHeaderTitle:@"Title" cancelButtonTitle:@"Cancel" confirmButtonTitle:@"Confirm" switchButtonTitle:@"Don't ask me"];
    picker.delegate = self;
    picker.dataSource = self;
    [picker showAndSelectedIndex:3];
}
@end

File: DYAlertPickerViewDemo-Swift/ViewController.swift
//
//  ViewController.swift
//  DYAlertPickerViewDemo-Swift
//
//  Created by danny on 2016/1/7.
//  Copyright © 2016年 danny. All rights reserved.
//

import UIKit

class ViewController: UIViewController, DYAlertPickViewDelegate, DYAlertPickViewDataSource {
    var items = Array<String>()
    var alertView:DYAlertPickView = DYAlertPickView()
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
 
    func pickerview(pickerView: DYAlertPickView!, titleForRow row: Int) -> NSAttributedString! {
        let str = NSAttributedString(string: items[row])
        return str
    }
    func numberOfRowsInPickerview(pickerView: DYAlertPickView!) -> Int {
        return items.count
    }
    
    
    func pickerview(pickerView: DYAlertPickView!, didConfirmWithItemAtRow row: Int) {
        print(items[row]," didConfirm");
    }
    
    func pickerviewDidClickCancelButton(pickerView: DYAlertPickView!) {
        print("Canceled");
    }
    
    func pickerviewDidClickSwitchButton(pickerView: DYAlertPickView!, switchButton: UISwitch!) {
        if (switchButton.on) {
            print("On")
        }else {
            print("Off")
        }
    }
    
    @objc func alertpickerviewStateOfSwitchButton() -> Bool {
        return true;
    }
    
    func showAlertPickerView(sender:UIButton ) {
        self.alertView = DYAlertPickView(headerTitle: "Title", cancelButtonTitle: "Cancel", confirmButtonTitle: "Confirm", switchButtonTitle: "Don't ask me")
        self.alertView.delegate = self;
        self.alertView.dataSource = self;
        items = ["Item 1", "Item 2", "Item 3", "Item 4", "Item 5"]
        self.alertView.showAndSelectedIndex(3)
    }
}

